server/test/unit/server/webservices/views/test_repositories.py

```diff
@@ -0,0 +1,1977 @@
+    @mock.patch('pulp.server.webservices.views.repositories.reverse')
+    def test__process_repos_minimal(self, mock_reverse, mock_convert_date):
+        """
+        Test _process_repos without optional args, assert that processing was called for each repo.
+        """
+
+        mock_reverse.return_value = '/mock/path/'
+        mock_repos = [{'id': 'mock1'}, {'id': 'mock2'}]
+        repos_view = ReposView()
+        repos_view._process_repos(mock_repos, False, False)
+        mock_convert_date.assert_has_calls(
+            [mock.call({'id': 'mock1', '_href': '/mock/path/'}),
+             mock.call({'id': 'mock2', '_href': '/mock/path/'})]
+        )
+
+    @mock.patch('pulp.server.webservices.views.repositories.manager_factory')
+    @mock.patch('pulp.server.webservices.views.repositories._merge_related_objects')
+    @mock.patch('pulp.server.webservices.views.repositories._convert_repo_dates_to_strings')
+    @mock.patch('pulp.server.webservices.views.repositories.reverse')
+    def test__process_repos_details(self, mock_reverse, mock_convert_date, mock_merge,
+                                    mock_factory):
+        """
+        Test that importer and distributor options cause the expected merges to happen.
+        """
+
+        mock_reverse.return_value = '/mock/path/'
+        mock_repos = [{'id': 'mock1'}, {'id': 'mock2'}]
+        repos_view = ReposView()
+        repos_view._process_repos(mock_repos, True, True)
+        mock_merge.assert_has_calls(
+            [mock.call('importers', mock_factory.repo_importer_manager(), mock_repos),
+             mock.call('distributors', mock_factory.repo_distributor_manager(), mock_repos)]
+        )
+        mock_convert_date.assert_has_calls(
+            [mock.call({'id': 'mock1', '_href': '/mock/path/'}),
+             mock.call({'id': 'mock2', '_href': '/mock/path/'})]
+        )
+
+    @mock.patch('pulp.server.webservices.views.repositories._convert_repo_dates_to_strings')
+    @mock.patch('pulp.server.webservices.views.repositories.reverse')
+    def test__process_repos_scratchpad(self, mock_reverse, mock_convert_date):
+        """
+        Test that scratchpad field has been removed.
+        """
+
+        mock_reverse.return_value = '/mock/path/'
+        mock_repos = [{'id': 'mock1', 'scratchpad': 'should be removed'}, {'id': 'mock2'}]
+        repos_view = ReposView()
+        repos_view._process_repos(mock_repos, False, False)
+        mock_convert_date.assert_has_calls(
+            [mock.call({'id': 'mock1', '_href': '/mock/path/'}),
+             mock.call({'id': 'mock2', '_href': '/mock/path/'})]
+        )
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_READ())
+    @mock.patch('pulp.server.webservices.views.repositories.ReposView._process_repos')
+    @mock.patch('pulp.server.webservices.views.repositories.RepoModel.get_collection')
```

Comment:
i think it would be better if you would also mock the generate_json_response_with_pulp_encoder and then assert that the response was called with proper expected content